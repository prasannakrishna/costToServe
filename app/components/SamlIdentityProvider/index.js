/**
 * Copyright © 2018, JDA Software Group, Inc. ALL RIGHTS RESERVED.
 * <p>
 * This software is the confidential information of JDA Software, Inc., and is licensed
 * as restricted rights software. The use,reproduction, or disclosure of this software
 * is subject to restrictions set forth in your license agreement with JDA.
 */
/**
 *
 * SamlIdentityProvider
 *
 */

import React from 'react';
import MuiThemeProvider from '@material-ui/core/styles/MuiThemeProvider';
import { DarkTheme } from '@jda/luminate-platform-ui';
import styled from 'styled-components';
import buttonImage from './sign-in-button.png';
import jdaLogo from './logo_login_jda.png';

const sectionCenteredStyle = {
  fontFamily: 'Roboto',
  position: 'absolute',
  top: '50%',
  // left:  '50%',
  textAlign: 'center',
  width: '100%',
  padding: '0 20px',
  boxSizing: 'border-box',
};

const copyrightStyle = {
  textAlign: 'center',
  width: '100%',
  padding: '0 20px',
  boxSizing: 'border-box',
  position: 'absolute',
  bottom: '30px',
  //  flexDirection: 'column',
  //  left: '50%'
  // transform: 'translateX(-50%)'
};
const applicationTitleStyle = {
  fontFamily: 'Roboto',
  fontSize: '28px',
  color: '#FAFAFA',
  fontWeight: '300',
};

const loginButtonWrapper = {
  marginTop: '10px',
};

const copyrightMsg = {
  fontFamily: 'Roboto',
  fontSize: '10px',
  color: 'white',
  textAlign: 'center',
  opacity: '0.55',
};

const msgAreaStyle = {
  width: '100%',
  maxWidth: '600px',
  borderRadius: '3px',
  boxSizing: 'border-box',
  padding: '0 20px',
  height: '100%',
};

const AppWrapper = styled.div``;

const AppFrame = styled.div`
  margin-top: 0px;
  overflow-x: hidden;
`;
/* eslint-disable react/prefer-stateless-function */
class SamlIdentityProvider extends React.Component {
  render() {
    return (
      <MuiThemeProvider theme={DarkTheme}>
        <AppWrapper
          style={{
            backgroundColor: DarkTheme.palette.primary[200],
            height: '100vh',
          }}
        >
          <AppFrame>
            <div style={msgAreaStyle}>
              <div className="section centered" style={sectionCenteredStyle}>
                <div
                  className="application-title"
                  style={applicationTitleStyle}
                >
                  JDA Luminate YODA Platform
                </div>
                <div
                  className="login-button-wrapper"
                  style={loginButtonWrapper}
                >
                  {/* <Link to='/loginform' > */}
                  <a href="/auth">
                    <img
                      src={buttonImage}
                      width="256"
                      height="47"
                      alt="sign in button"
                    />
                  </a>
                  {/* </Link> */}
                </div>
              </div>

              <div className="section copyright-section" style={copyrightStyle}>
                <div>
                  <img src={jdaLogo} alt="JDA plan to deliver" />
                </div>
                <span className="copyright-msg" style={copyrightMsg}>
                  Copyright © 2019, JDA Software Group, Inc. ALL RIGHTS
                  RESERVED.
                  <br />
                  This software is the confidential information of JDA Software,
                  Inc., and is licensed as restricted rights software.The use,{' '}
                  <br />
                  reproduction, or disclosure of this software is subject to
                  restrictions set forth in your license agreement with JDA.
                </span>
              </div>
            </div>
          </AppFrame>
        </AppWrapper>
      </MuiThemeProvider>
    );
  }
}

SamlIdentityProvider.propTypes = {};

export default SamlIdentityProvider;
