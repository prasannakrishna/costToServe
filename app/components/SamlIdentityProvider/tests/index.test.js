/**
 * Copyright Â© 2018, JDA Software Group, Inc. ALL RIGHTS RESERVED.
 * <p>
 * This software is the confidential information of JDA Software, Inc., and is licensed
 * as restricted rights software. The use,reproduction, or disclosure of this software
 * is subject to restrictions set forth in your license agreement with JDA.
 */
import React from 'react';
import { shallow } from 'enzyme';
import SamlIdentityProvider from '../index';

describe('<SamlIdentityProvider />', () => {
  it('should be rendered correctly.', () => {
    const componentWrapper = shallow(
      <SamlIdentityProvider />
    );
    expect(componentWrapper).toMatchSnapshot();
  });
});
