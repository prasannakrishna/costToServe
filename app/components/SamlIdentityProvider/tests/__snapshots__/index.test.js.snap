// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SamlIdentityProvider /> should be rendered correctly. 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SamlIdentityProvider />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="section centered"
          style={
            Object {
              "boxSizing": "border-box",
              "fontFamily": "Roboto",
              "padding": "0 20px",
              "position": "absolute",
              "textAlign": "center",
              "top": "50%",
              "width": "100%",
            }
          }
        >
          <div
            className="application-title"
            style={
              Object {
                "color": "#FAFAFA",
                "fontFamily": "Roboto",
                "fontSize": "28px",
                "fontWeight": "300",
              }
            }
          >
            JDA Luminate Control Tower
          </div>
          <div
            className="login-button-wrapper"
            style={
              Object {
                "marginTop": "10px",
              }
            }
          >
            <a
              href="/auth"
            >
              <img
                alt="sign in button"
                height="47"
                src="IMAGE_MOCK"
                width="256"
              />
            </a>
          </div>
        </div>,
        <div
          className="section copyright-section"
          style={
            Object {
              "bottom": "30px",
              "boxSizing": "border-box",
              "padding": "0 20px",
              "position": "absolute",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          <div>
            <img
              alt="JDA plan to deliver"
              src="IMAGE_MOCK"
            />
          </div>
          <span
            className="copyright-msg"
            style={
              Object {
                "color": "white",
                "fontFamily": "Roboto",
                "fontSize": "10px",
                "opacity": "0.55",
                "textAlign": "center",
              }
            }
          >
            Copyright © 2018, JDA Software Group, Inc. ALL RIGHTS RESERVED.
            <br />
            This software is the confidential information of JDA Software, Inc., and is licensed as restricted rights software.The use, 
            <br />
            reproduction, or disclosure of this software is subject to restrictions set forth in your license agreement with JDA.
          </span>
        </div>,
      ],
      "style": Object {
        "borderRadius": "3px",
        "boxSizing": "border-box",
        "height": "100%",
        "maxWidth": "600px",
        "padding": "0 20px",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="application-title"
              style={
                Object {
                  "color": "#FAFAFA",
                  "fontFamily": "Roboto",
                  "fontSize": "28px",
                  "fontWeight": "300",
                }
              }
            >
              JDA Luminate Control Tower
            </div>,
            <div
              className="login-button-wrapper"
              style={
                Object {
                  "marginTop": "10px",
                }
              }
            >
              <a
                href="/auth"
              >
                <img
                  alt="sign in button"
                  height="47"
                  src="IMAGE_MOCK"
                  width="256"
                />
              </a>
            </div>,
          ],
          "className": "section centered",
          "style": Object {
            "boxSizing": "border-box",
            "fontFamily": "Roboto",
            "padding": "0 20px",
            "position": "absolute",
            "textAlign": "center",
            "top": "50%",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "JDA Luminate Control Tower",
              "className": "application-title",
              "style": Object {
                "color": "#FAFAFA",
                "fontFamily": "Roboto",
                "fontSize": "28px",
                "fontWeight": "300",
              },
            },
            "ref": null,
            "rendered": "JDA Luminate Control Tower",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="/auth"
              >
                <img
                  alt="sign in button"
                  height="47"
                  src="IMAGE_MOCK"
                  width="256"
                />
              </a>,
              "className": "login-button-wrapper",
              "style": Object {
                "marginTop": "10px",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="sign in button"
                  height="47"
                  src="IMAGE_MOCK"
                  width="256"
                />,
                "href": "/auth",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "sign in button",
                  "height": "47",
                  "src": "IMAGE_MOCK",
                  "width": "256",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "a",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <img
                alt="JDA plan to deliver"
                src="IMAGE_MOCK"
              />
            </div>,
            <span
              className="copyright-msg"
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Roboto",
                  "fontSize": "10px",
                  "opacity": "0.55",
                  "textAlign": "center",
                }
              }
            >
              Copyright © 2018, JDA Software Group, Inc. ALL RIGHTS RESERVED.
              <br />
              This software is the confidential information of JDA Software, Inc., and is licensed as restricted rights software.The use, 
              <br />
              reproduction, or disclosure of this software is subject to restrictions set forth in your license agreement with JDA.
            </span>,
          ],
          "className": "section copyright-section",
          "style": Object {
            "bottom": "30px",
            "boxSizing": "border-box",
            "padding": "0 20px",
            "position": "absolute",
            "textAlign": "center",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="JDA plan to deliver"
                src="IMAGE_MOCK"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "JDA plan to deliver",
                "src": "IMAGE_MOCK",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Copyright © 2018, JDA Software Group, Inc. ALL RIGHTS RESERVED.",
                <br />,
                "This software is the confidential information of JDA Software, Inc., and is licensed as restricted rights software.The use, ",
                <br />,
                "reproduction, or disclosure of this software is subject to restrictions set forth in your license agreement with JDA.",
              ],
              "className": "copyright-msg",
              "style": Object {
                "color": "white",
                "fontFamily": "Roboto",
                "fontSize": "10px",
                "opacity": "0.55",
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": Array [
              "Copyright © 2018, JDA Software Group, Inc. ALL RIGHTS RESERVED.",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              "This software is the confidential information of JDA Software, Inc., and is licensed as restricted rights software.The use, ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              "reproduction, or disclosure of this software is subject to restrictions set forth in your license agreement with JDA.",
            ],
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="section centered"
            style={
              Object {
                "boxSizing": "border-box",
                "fontFamily": "Roboto",
                "padding": "0 20px",
                "position": "absolute",
                "textAlign": "center",
                "top": "50%",
                "width": "100%",
              }
            }
          >
            <div
              className="application-title"
              style={
                Object {
                  "color": "#FAFAFA",
                  "fontFamily": "Roboto",
                  "fontSize": "28px",
                  "fontWeight": "300",
                }
              }
            >
              JDA Luminate Control Tower
            </div>
            <div
              className="login-button-wrapper"
              style={
                Object {
                  "marginTop": "10px",
                }
              }
            >
              <a
                href="/auth"
              >
                <img
                  alt="sign in button"
                  height="47"
                  src="IMAGE_MOCK"
                  width="256"
                />
              </a>
            </div>
          </div>,
          <div
            className="section copyright-section"
            style={
              Object {
                "bottom": "30px",
                "boxSizing": "border-box",
                "padding": "0 20px",
                "position": "absolute",
                "textAlign": "center",
                "width": "100%",
              }
            }
          >
            <div>
              <img
                alt="JDA plan to deliver"
                src="IMAGE_MOCK"
              />
            </div>
            <span
              className="copyright-msg"
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Roboto",
                  "fontSize": "10px",
                  "opacity": "0.55",
                  "textAlign": "center",
                }
              }
            >
              Copyright © 2018, JDA Software Group, Inc. ALL RIGHTS RESERVED.
              <br />
              This software is the confidential information of JDA Software, Inc., and is licensed as restricted rights software.The use, 
              <br />
              reproduction, or disclosure of this software is subject to restrictions set forth in your license agreement with JDA.
            </span>
          </div>,
        ],
        "style": Object {
          "borderRadius": "3px",
          "boxSizing": "border-box",
          "height": "100%",
          "maxWidth": "600px",
          "padding": "0 20px",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="application-title"
                style={
                  Object {
                    "color": "#FAFAFA",
                    "fontFamily": "Roboto",
                    "fontSize": "28px",
                    "fontWeight": "300",
                  }
                }
              >
                JDA Luminate Control Tower
              </div>,
              <div
                className="login-button-wrapper"
                style={
                  Object {
                    "marginTop": "10px",
                  }
                }
              >
                <a
                  href="/auth"
                >
                  <img
                    alt="sign in button"
                    height="47"
                    src="IMAGE_MOCK"
                    width="256"
                  />
                </a>
              </div>,
            ],
            "className": "section centered",
            "style": Object {
              "boxSizing": "border-box",
              "fontFamily": "Roboto",
              "padding": "0 20px",
              "position": "absolute",
              "textAlign": "center",
              "top": "50%",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "JDA Luminate Control Tower",
                "className": "application-title",
                "style": Object {
                  "color": "#FAFAFA",
                  "fontFamily": "Roboto",
                  "fontSize": "28px",
                  "fontWeight": "300",
                },
              },
              "ref": null,
              "rendered": "JDA Luminate Control Tower",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="/auth"
                >
                  <img
                    alt="sign in button"
                    height="47"
                    src="IMAGE_MOCK"
                    width="256"
                  />
                </a>,
                "className": "login-button-wrapper",
                "style": Object {
                  "marginTop": "10px",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="sign in button"
                    height="47"
                    src="IMAGE_MOCK"
                    width="256"
                  />,
                  "href": "/auth",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "sign in button",
                    "height": "47",
                    "src": "IMAGE_MOCK",
                    "width": "256",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "a",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <img
                  alt="JDA plan to deliver"
                  src="IMAGE_MOCK"
                />
              </div>,
              <span
                className="copyright-msg"
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Roboto",
                    "fontSize": "10px",
                    "opacity": "0.55",
                    "textAlign": "center",
                  }
                }
              >
                Copyright © 2018, JDA Software Group, Inc. ALL RIGHTS RESERVED.
                <br />
                This software is the confidential information of JDA Software, Inc., and is licensed as restricted rights software.The use, 
                <br />
                reproduction, or disclosure of this software is subject to restrictions set forth in your license agreement with JDA.
              </span>,
            ],
            "className": "section copyright-section",
            "style": Object {
              "bottom": "30px",
              "boxSizing": "border-box",
              "padding": "0 20px",
              "position": "absolute",
              "textAlign": "center",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="JDA plan to deliver"
                  src="IMAGE_MOCK"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "JDA plan to deliver",
                  "src": "IMAGE_MOCK",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Copyright © 2018, JDA Software Group, Inc. ALL RIGHTS RESERVED.",
                  <br />,
                  "This software is the confidential information of JDA Software, Inc., and is licensed as restricted rights software.The use, ",
                  <br />,
                  "reproduction, or disclosure of this software is subject to restrictions set forth in your license agreement with JDA.",
                ],
                "className": "copyright-msg",
                "style": Object {
                  "color": "white",
                  "fontFamily": "Roboto",
                  "fontSize": "10px",
                  "opacity": "0.55",
                  "textAlign": "center",
                },
              },
              "ref": null,
              "rendered": Array [
                "Copyright © 2018, JDA Software Group, Inc. ALL RIGHTS RESERVED.",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                "This software is the confidential information of JDA Software, Inc., and is licensed as restricted rights software.The use, ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                "reproduction, or disclosure of this software is subject to restrictions set forth in your license agreement with JDA.",
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
